packages/authentication/docs/authentication-action.md

```diff
@@ -0,0 +1,32 @@
+### Auth action
+
+```ts
+async action(request: Request): Promise<UserProfile | undefined> {
+    const strategy = await this.getStrategy();
+    // read the action metadata from endpoint's auth metadata like 
+    // `@authenticate('strategy_name', {actions: 'authenticate'})`
+    // type ActionType = 'authenticate' | 'ensureLoggedIn'
+    const action = await this.getAction();
```

Comment:
Maybe we don't need to differentiate between `login` and `verify` actions. Please note `Passport` just uses different strategies, for example:

- Local strategy for username/password based login
- BearTokenStrategy for token verification

Reply:
I see, I was coupling them into one strategy.